### PR TITLE
Fix fiber logs on non existing endpoints

### LINKED ISSUE #35: Unhandled error for not exposed endpoints
The function RequestFiberMiddlewareLogger prints a log before and after handling the endpoints.

The error occur when someone calls endpoints not exposed by the service. The service returns correctly a 404 not found while middleware logs a statusCode 200 with empty body:
```
{"host":{"hostname":"localhost"},"http":{"request":{"method":"GET","userAgent":{"original":"PostmanRuntime/7.29.2"}},"response":{"statusCode":200,"body":{"bytes":0}}},"level":30,"msg":"request completed","reqId":"c3975899-1319-418a-a34f-a683b9126d5e","responseTime":0,"time":1677144605114,"url":{"path":"/fake/endpoint"}}
```

With some debugging I found that, in this case, the error at line 80 is evaluated with the correct statusCode but it's not handled:
https://github.com/mia-platform/glogger/blob/0ce2ce130fcdd229c05d2bc88af86e88951a2c32/middleware/fibermiddleware.go#L80

### PR DESCRIPTION
**Pull request by malta895**
Fixes #35

### CODE REVIEW

File: middleware/fibermiddleware_test.go [hunk 1]
```diff
@@ -53,6 +55,109 @@ func testMockFiberMiddlewareInvocation(handler fiber.Handler, requestID string,
...
+		const statusCode = 200
+		const requestID = "my-req-id"
+		const reqPath = "/my-req"
+		var requestPathWithoutPort = fmt.Sprintf("http://%s%s", hostname, reqPath)
+
+		logger, _ := test.NewNullLogger()
+		hook := testMockFiberMiddlewareInvocation(func(c *fiber.Ctx) error {
+			c.Status(statusCode)
+			return nil
+		}, requestID, logger, requestPathWithoutPort)
```
Comment (malta895) [inline middleware/fibermiddleware_test.go:69]: @davidebianchi @fredmaggiowski Hi! I can't understand the purpose of this test: it always fail in 404, and I think is because the `requestPathWithoutPort` is not a valid path to be set to the router, and the requests fail.

The bug I'm trying to fix hid this failure, because it caused every request to a non existing route to log a 200 status, so this test didn't fail.

File: middleware/fibermiddleware.go [hunk 2]
```diff
@@ -78,8 +78,13 @@ func RequestFiberMiddlewareLogger(logger *logrus.Logger, excludedPrefix []string
 
 		logBeforeHandler(fiberLoggingContext)
 		err := fiberCtx.Next()
-		logAfterHandler(fiberLoggingContext, start)
-
+		if err != nil {
+			if fiberErr, ok := err.(*fiber.Error); ok {
+				logAfterHandler(fiberLoggingContext, start, fiberErr)
```
Comment (fredmaggiowski) [inline middleware/fibermiddleware.go:83]: can't we invoke the logAfterHandler only once always passing the err?

```
err := fiberCtx.Next()
logAfterHandler(fiberLoggingContext, start, err)
return err
```

File: middleware/fibermiddleware_test.go (hunk 1 above)
Comment (fredmaggiowski) [reply middleware/fibermiddleware_test.go:69]: I think we can change this test if it does not make sense; I don't recall the specifications for the fiber integration, it was developed by @mattia-fumo maybe he can give us some more insights on this

File: middleware/logmiddleware.go [hunk 3]
```diff
@@ -93,19 +94,32 @@ func logBeforeHandler(ctx loggingContext) {
 	}).Trace("incoming request")
 }
 
-func logAfterHandler(ctx loggingContext, startTime time.Time) {
-	glogger.Get(ctx.Context()).WithFields(logrus.Fields{
+func logAfterHandler(ctx loggingContext, startTime time.Time, fiberErr *fiber.Error) {
```
Comment (fredmaggiowski) [inline middleware/logmiddleware.go:97]: Since this function is not framework specific, I'd accept a generic `error` interface and then do type assertion at line 107 if needed

**Review by fredmaggiowski — CHANGES_REQUESTED** [review]

File: middleware/fibermiddleware_test.go (hunk 1 above)
Comment (mattia-fumo) [reply middleware/fibermiddleware_test.go:69]: This test just check the hostname parsing and logging behavior in the specific case where the request url has no port specified, why do you think its useless or buggy?

File: middleware/logmiddleware.go (hunk 3 above)
Comment (malta895) [reply middleware/logmiddleware.go:97]: I agree, done

File: middleware/fibermiddleware_test.go (hunk 1 above)
Comment (mattia-fumo) [reply middleware/fibermiddleware_test.go:69]: I can see it now. I would rather say that we should strip the request path in `testMockFiberMiddlewareInvocation` and register the `app.Get` on it. What do you think?

Comment (mattia-fumo) [reply middleware/fibermiddleware_test.go:69]: I can see it now. This test needs an hostname in the request, thats why it uses an absolute URL instead of a relative path. Maybe we could always compose a request with an absolute path [here](https://github.com/mia-platform/glogger/pull/37/files#diff-d8990d7a3bab349cf5a6ed0b933a59e463c6632baa708cd7f885c04c9f00dd44R25) to make the call even more realistic.
Like this:
```go
	req := httptest.NewRequest(http.MethodGet, fmt.Sprintf("http://example.com%s", requestPath), nil)
```
And remove [this line](https://github.com/mia-platform/glogger/pull/37/files#diff-d8990d7a3bab349cf5a6ed0b933a59e463c6632baa708cd7f885c04c9f00dd44R63).
Do you agree?

File: middleware/logmiddleware.go [hunk 4]
```diff
@@ -93,19 +94,33 @@ func logBeforeHandler(ctx loggingContext) {
...
+	if err != nil {
+		if fiberErr, ok := err.(*fiber.Error); ok {
+			res = &Response{
+				StatusCode: fiberErr.Code,
+				Body: map[string]interface{}{
+					"bytes": len(fiberErr.Error()),
+				},
+			}
+		}
+	}
```
Comment (mattia-fumo) [inline middleware/logmiddleware.go:115]: This two could be unified into: `err != nil && ok`, because the type cast operator is safe on `nil` values.

Comment (malta895) [reply middleware/logmiddleware.go]: done!

File: middleware/fibermiddleware_test.go (hunk 1 above)
Comment (malta895) [reply middleware/fibermiddleware_test.go]: Yes I agree, done. I also fixed a bug with the `ip` test in cb80b553031d752aa2405f6988f54a48a7f31e1f, it was declared as empty string, and the test were rewriting on it without using the value after. I just redeclared it as `const` making sure it is not overwritten anywhere.

**Review by fredmaggiowski — APPROVED** [review]

File: middleware/fibermiddleware_test.go (hunk 1 above)
Comment (mattia-fumo) [reply middleware/fibermiddleware_test.go]: lgtm
